pkg/bzz/address.go

```diff
@@ -114,21 +110,52 @@ func (a *Address) Equal(b *Address) bool {
 		return a == b
 	}
 
-	return a.Overlay.Equal(b.Overlay) && multiaddrEqual(a.Underlay, b.Underlay) && bytes.Equal(a.Signature, b.Signature) && bytes.Equal(a.Nonce, b.Nonce)
+	return a.Overlay.Equal(b.Overlay) && IsUnderlayEqual(a.Underlay, b.Underlay) && bytes.Equal(a.Signature, b.Signature) && bytes.Equal(a.Nonce, b.Nonce)
 }
 
-func multiaddrEqual(a, b ma.Multiaddr) bool {
-	if a == nil || b == nil {
-		return a == b
+func IsAtLeastOneUnderlayEqual(a, b []ma.Multiaddr) bool {
+	if len(a) == 0 || len(b) == 0 {
+		return false
+	}
+	for i := 0; i < len(a); i++ {
+		for j := 0; j < len(b); j++ {
+			if a[i] != nil && b[j] != nil && a[i].Equal(b[j]) {
+				return true
+			}
+		}
 	}
+	return false
+}
 
-	return a.Equal(b)
+func IsUnderlayEqual(a, b []ma.Multiaddr) bool {
```

Comment:
Would name `AreUnderlaysEqual` (or something similar with plurals) make more sense as it compares lists?